client/packages/system/src/Patient/api/hooks/utils/useLinkPatientToStore.ts

```diff
@@ -0,0 +1,9 @@
+import { useMutation } from '@openmsupply-client/common';
+import { usePatientApi } from '../utils/usePatientApi';
+
+export const useLinkPatientToStore = () => {
+  const api = usePatientApi();
+  return {
+    ...useMutation((nameId: string) => api.linkPatientToStore(nameId)),
+  };
```

Comment:
Though, I would also invalidate the query cache, like so:

```
import { useMutation, useQueryClient } from '@openmsupply-client/common';
import { usePatientApi } from '../utils/usePatientApi';

export const useLinkPatientToStore = () => {
  const api = usePatientApi();
  const queryClient = useQueryClient();

  return useMutation((nameId: string) => api.linkPatientToStore(nameId), {
    onSuccess: () => {
      queryClient.invalidateQueries(api.keys.base());
    },
  });
};
```

Currently - after fetching, the patient which I've searched for still shows as 'remote' and I can try to download them again. If you invalidate the cache then they show as local immediately.

Reply:
Will check why this is the case, but I don't really see why to invalidate here because linking a patient shouldn't immediately change anything locally. We would still need to sync before there are any changes. Maybe this is what happen in the background? Do we have a mechanism to invalidate something after sync? e.g. throw away all cached queries?

Reply:
the first time after I made this change it worked perfectly - after that it wasn't as reliable. so there must be a sync delay too!

the currently display list of patients won't change if you don't invalidate the cache.. but you are right. better to hang it off the back of the manual sync completing?